### DISCUSSION TITLE
Year range for higher-level processing

### DISCUSSION THREAD

**Discussion by corneliussenf** [davidfrantz/force#79] [Ideas]
Hi all!

I king of experienced an unexpected behavior, which makes me wonder whether the higher-level processing parameter file could be improved in usability. 

When I enter a multi-year date range in the 'PROCESSING TIMEFRAME' section (e.g., DATE_RANGE = 1984-01-01 2020-12-31) but specify only one year in the 'BAP' section (e.g., YEAR_TARGET = 2018 and YEAR_NUM = 0) I would expect the composite to be build with images from 2018 only, correct? The composite, however, uses images from the full time series (i.e., 1984 to 2020). Is this intended behavior? It works if I limit the DATE_RANGE argument to only include images from 2018. This is kind of annoying, because I need to change both arguments (DATE_RANGE and YEAR_TARGET) when creating composites for several years.

Another feature that would be really helpful and realties to this would be an option to specify several target years (e.g., YEAR_TARGET = 2017 2018 2019), resulting in three composites. Then I could create composites for several years from one parameter file without changing the target year (and processing time frame) manually.

Let me know whether my explanation is unclear oder whether I misunderstood something.

Cornelius

**Comment by davidfrantz** [comment]
Hi Cornelius, 

this is actually by design.

The ``YEAR_NUM`` parameter is used to compute the annual weight as described in Frantz et al. 2017 (RSE).
The reason why other years are used, too: Imagine building a composite centered in January, you surely would want data from ``YEAR_TARGET`` - 1, too.

``DATE_RANGE`` specifies the data to be read, ``YEAR_TARGET, YEAR_NUM, and Y_FACTOR`` specify how the data are weighted.

You can try weighting the Year score more (``SCORE_YEAR_WEIGHT``) than the other scores to mitigate this.
Both from a logical and performance (I/O) POV, it makes sense to narrow down the ``DATE_RANGE``.

For the looping: we have [force-magic-parameters](https://force-eo.readthedocs.io/en/latest/components/auxilliary/magic-parameters.html), which kind of allows you to do this.

Cheers,
David

**Reply by corneliussenf** [reply]
Ok, this clarifies things. Thanks David. Will also check out the force-magic-parmater function. Didn't know about it!
